### PR TITLE
Structured metadata: Changes to ad-hoc variable doesn't run detected_fields 

### PR DESCRIPTION
**Pull request by gtk-grafana**
Fixes: https://github.com/grafana/explore-logs/issues/847

### CODE REVIEW

File: tests/exploreServicesBreakDown.spec.ts [hunk 1]
```diff
@@ -403,19 +402,15 @@ test.describe('explore services breakdown page', () => {
...
-
     // Panel on the top should not
     await expect(page.getByTestId(/data-testid Panel header/).first()).not.toBeInViewport();
-
     // Wait for a bit for the requests to be made
     await page.waitForTimeout(250);
-
     // if this flakes we could just assert that it's greater then 3
-    expect(requestCount).toEqual(14);
+    expect(requestCount).toEqual(17);
```
Comment (gtk-grafana) [inline tests/exploreServicesBreakDown.spec.ts:413]: Don't know why there is an extra row now, this shouldn't have anything to do with the functional change in this PR, I think it's a consequence of the updates to 11.3

File: tests/fixtures/explore.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ export class ExplorePage {
   }
 
   async scrollToBottom() {
-    const main = this.page.locator('main#pageContent');
+    const main = this.page.locator('html');
```
Comment (gtk-grafana) [inline tests/fixtures/explore.ts:85]: scroll container appears to have changed in 11.3

File: tests/fixtures/explore.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export class ExplorePage {
   }
 
   async setDefaultViewportSize() {
-    await this.page.setViewportSize({ width: 1280, height: 720 });
+    await this.page.setViewportSize({ width: 1280, height: 680 });
```
Comment (gtk-grafana) [inline tests/fixtures/explore.ts:47]: window height also changed a bit in 11.3, which had the consequence of adding more lazy loaded elements into the viewport, bumping down the height a tiny bit so we don't have to update a bunch of assertions.

**Comment by matyax** [comment]
I'm seeing that it correctly runs with filter changes except with patterns.

Uploading Test.mov…

**Comment by gtk-grafana** [comment]
@matyax great catch, I was initially thinking patterns should be fine because the patterns endpoint doesn't take detected fields, but it turns out detected_fields does take patterns!

Should be fixed now

**Review by matyax — APPROVED** [review]
GJ!

**Pull request merged**